sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -274,77 +277,80 @@ AccessToken TokenCredentialImpl::ParseToken(
     }
   }
 
-  if (expiresOnPropertyName.empty())
+  if (expiresOnPropertyNames.front().empty())
```

Comment:
We are picking the first one here to check. Why? It's a bit brittle to rely on the order in which the expiresOn property names vector is constructed. If we must rely on the order, there should be a comment at construction time highlighting that dependnecy.

Reply:
This is private, so I thought we could be less vandal-proof about the contract. But let me to update it.

Reply:
I updated the code, now we check the entire vector for non-empty items.